### ISSUE TITLE
[Feature]:Integrating its code into nvdaa and developing more flexible braille keys for people who lack arms or fingers.

### ISSUE THREAD

**Issue by levilibegas** [nvdaes/pcKbBrl#9]
### Problem

I wonder if this addon could be integrated in core NVDA, it look has been but not the last version. I wonder if it could be made more flexible the assignation of dots.

### Suggestion

Hi, I just started a issue in nvda, I am Ivan. I firstly wanted to thank for this lovely addon. I have seen it looks implemented in nvda however the nvda does not havethe last changes of this Braille Keyboard as for instance, creating the key combination after certain time...I wonder if it would be doable to reassing the eight braille dots and the dots that eliminate pressed dots in one handed typing and aswell the keys to let the key being made. I have seen people who would rather type in vertical rows or that would prefer to have unused keys dead so they wluld not accidentally press them. I really apreciateit a lot. Thanks for all.

### Additional context

I have a friend who would love to test any possiblechanges. Change.

**Comment by nvdaes** [comment]
Hello, thanks for your interest in this add-on and happy new year 2023!
Currently, it's possible to choose the keys to confirm and cancel in One hand mode.
In short, what you want is:
- Make possible for not assigned keys to be disabled.
- Make possible to choose the key for 8 dot, mostly useful for keyboards not supported.
- Make possible assignment of any dots, or maybe different layouts? For this last point I think that providing use cases or practical reasons that justify this maybe helpful. For now, I'll start working in the two first items. I'll provide versions of the add-on for testing.
I'll work in the add-on for now. If some day this feature is to be included in NVDA's core, I may help if appropriate.

**Comment by nvdaes** [comment]
After thinking with more detail, for now I'll try to make configurable dots and dead keys for one hand mode. IFor me it's not clear if you would like to configure dots also for two hand writing. I think not, since typing with vertical layout seems not very comfortable. For now, lets narrow this issue to one handmode. If you need more features, you can open a new issue.

**Comment by levilibegas** [comment]
Hi, thanks... Yes, vertical typing would be suitable for one handed mode. I don't know how I could try these changes, as NVDA already has keyboard braille addon embeded in it and does not seem to let me use the updated addon... I am not sure if I am saying it clear... My friend who lacks his right hand would love to talk in Spanish so he would explain his concerns so the three of us would better discuss it, my mail is ***@***.*** thanks. Enviado desde mi iPhoneEl 4 ene 2023, a las 6:38, Noelia Ruiz Martínez ***@***.***> escribió:﻿
After thinking with more detail, for now I'll try to make configurable dots and dead keys for one hand mode. IFor me it's not clear if you would like to configure dots also for two hand writing. I think not, since typing with vertical layout seems not very comfortable. For now, lets narrow this issue to one handmode. If you need more features, you can open a new issue.

**Comment by nvdaes** [comment]
Í enxpect to release a version for testing in a few hours.

Enviado desde mi iPhone

**Comment by nvdaes** [comment]
Please, feel free to unzip, install and test with this add-on. Go to NVDA's menu, Preferences submenu, Options, and explore the new options to configure dots to be used when typing with one hand, and the nullKeys edit box, to enter keys that should be ignored, for example, bz

[nvda-addon(5).zip](https://github.com/nvdaes/pcKbBrl/files/10345080/nvda-addon.5.zip)

**Comment by levilibegas** [comment]
Thanks, I am afraid I cannot install it as options are not shown. I am running latest snapshots of nvda that has your addon in it, however, even if I installed your addon I can only in settings one hand mode and speak dots as they are typed, Am I doing something wrong? thanks once more again. Enviado desde mi iPhoneEl 4 ene 2023, a las 15:55, Noelia Ruiz Martínez ***@***.***> escribió:﻿
Please, feel free to unzip, install and test with this add-on. Go to NVDA's menu, Preferences submenu, Options, and explore the new options to configure dots to be used when typing with one hand, and the nullKeys edit box, to enter keys that should be ignored, for example, bz
nvda-addon(5).zip

**Comment by nvdaes** [comment]
You may try again. Unzip the add-on and retry; You can also try to remove the old version before installing this one. Hope this helps.

**Comment by levilibegas** [comment]
Hi again, thanks, yes, I did try it all, I installed several nvda versions since the 2019 3 till the last development, I ran them portable and installed. It only tells me in brl keyboard pc addon's settings two options change one hand mode and dots as they are pressed in one hand mode. I wonder if there might be any sort of addong nvda changes in last version, or something I might do wrong? I deleted and installed all your addon version, development, stable, this very last one, no options were shown in its settings... I really apreciate a lot your help.Enviado desde mi iPhoneEl 4 ene 2023, a las 17:39, Noelia Ruiz Martínez ***@***.***> escribió:﻿
You may try again. Unzip the add-on and retry; You can also try to remove the old version before installing this one. Hope this helps.

**Comment by nvdaes** [comment]
If you want, send me a private e-mail and I can reply you with the add-on attached there, in case something is wrong in the download. I am spanish, in case you speak this language. Thanks. My e-mail is: ***@***.*** 
Enviado desde mi iPhone

**Comment by levilibegas** [comment]
I guess your mile was cut out, I will try to write here mine ivanonce@gmail.com

**Comment by levilibegas** [comment]
Thanks for taking your time to help  us, I am Spanish

**Comment by nvdaes** [comment]
Fixed in PR #10 
More work can be requested/done in other issues and pull requests.
